docs/reference/query-builders.md

```diff
@@ -0,0 +1,382 @@
+
+### `page()`
+
+```
+query.page(5)
+```
+
+The `page` builder is used to request a particular "page," or set, of results when the [`perPage` builder](#perpage) is also used. Page numbers start with `1` (there is no page zero).
+
+### `pageUrl()`
+
+```
+query.pageUrl(false)
+```
+
+The `pageUrl` query builder is set to `true` by default and controls whether to add the `._url` property to page documents in the query results. The builder can be set to `false` to disable the `._url` property on a particular query.
+
+The `._url` property will include a site prefix if applicable and is always better to use than the document's `slug` for a URL.
+
+### `permission()`
+
+```
+query.permission('edit')
+```
+
+The `permission` builder is used restrict returned documents based on the action name passed as an argument. Only documents on which the `req` object can take the named action are returned. For example, using `.permission('edit')` on the query will only return documents that the requesting user (via `req`) can edit.
+
+Valid action arguments include:
+- `'view'`: The `req` has permission to view the documents. This is the default.
+- `'edit'`: The `req` has permission to edit the documents.
+- `'publish'`: The `req` has permission to publish the documents.
+- `false`: Bypass any permission checks. This returns everything regardless of permission level. *Use this with caution.*
+
+In all cases, all of the returned docs are marked with `_edit: true` properties if the user associated with the request is allowed to edit the document. This is useful if you are fetching docs for viewing but also want to know which ones can be edited.
+
+### `perPage()`
+
+```
+query.perPage(20)
+```
+
+Using the `perPage` builder returns documents in sets of the number passed as an argument. This helps return documents in managable numbers and paginate the results, using the [`page` query builder](#page) to get specific set of results. This is usually easier than using `skip` and `limit` directly.
+
+### `previous()`
+
+```
+query.previous({
+  _id: 'cgwxzcdvc501b4v3ghxvxqgg1:en:published',
+  ...
+})
+```
+
+Passing a document object to the `previous` builder returns the document that precedes it in the current sort order. The document object argument needs to at least include the properties used by the current sort as well as the `_id`. The `_id` is used as a tiebreaker to avoid loops.
+
+### `project()`
+
+```
+query.project({
+  title: 1,
+  category: 1,
+  updatedAt: 1
+})
+```
+
+The `project` builders sets the [MongoDB projection](https://docs.mongodb.com/manual/tutorial/project-fields-from-query-results/), specifying the document properties to included in the returned documents. The argument should be an object with properties of desired field names set to `1` to include those fields.
+
```

Comment:
Apostrophe extends `project` in the following ways:

* `_url: 1` will add all properties required to generate an accurate `_url` property for each returned document.
* `_relationshipName: 1` will add the properties required to permit a field of type `relationship` to be loaded (in this case, the name of the field is `_relationshipName`), even though the related documents are not stored redundantly in the database.

Reply:
Negative projections, e.g. `body: 0`, are currently not supported.